src/common/applicationTags.js

```diff
@@ -0,0 +1,101 @@
+const constants = require('./constants');
+
+/**
+ * Metadata about your application represented as tags in Wavefront.
+ */
+class ApplicationTags {
+  /**
+   * Construct ApplicationTags.
+   * @param application - Application Name
+   * @param service - Service Name
+   * @param cluster - Cluster Name
+   * @param shard - Shard Name
+   * @param customTags - Array of arrays of custom tags
+   */
+  constructor({
+    application,
+    service,
+    cluster = null,
+    shard = null,
+    customTags = null
+  }) {
+    if (!application) {
+      throw `Missing "application" parameter in ApplicationTags!`;
+    }
+
+    if (!service) {
+      throw `Missing "service" parameter in ApplicationTags!`;
+    }
+
+    this._application = application;
+    this._service = service;
+    this._cluster = cluster;
+    this._shard = shard;
+    this._customTags = customTags;
+  }
+
+  /**
+   * Get Application Name.
+   * @returns {string}
+   */
+  get application() {
```

Comment:
there is no real private/public distinction in javascript, since all your properties are exposed anyway.  We generally just use this.application, and omit the getter.  For classes like this that are basically just data, we tend not to write classes at all, preferring object literals + transformation functions.

Reply:
The ApplicationTags class is implemented here to preserve states of the data. It is also consistent with the Java and Python SDKs. I'm wondering if it's better to switch to object literals + transformation functions? Otherwise I can replace the getter with the properties themselves :)

Reply:
hmm I guess there is no right answer. I think it depends on who will work on and maintain this code. I'd let authors/maintainers choose a coding style they're comfortable with. unless it's really against how Javascript should be written.

Reply:
I'm using the main wavefront codebase as a guideline, here.  We're gradually moving away from data-only classes, and we're also not implementing getters for anything.

Reply:
Changed class implementation to object and transformation function